src/main/java/com/google/devtools/build/lib/remote/ByteStreamUploader.java

```diff
@@ -323,7 +323,7 @@ private static String buildUploadResourceName(
               // level/algorithm, so we cannot know the expected committed offset
               long committedSize = committedOffset.get();
               long expected = chunker.getOffset();
-              if (!chunker.hasNext() && committedSize != expected) {
+              if (!chunker.hasNext() && committedSize != expected || (chunker.isCompressed() && committedSize == -1)) {
```

Comment:
Per the updated REAPI spec, 

```
... the request will terminate immediately without
error, and with a response whose `committed_size` is the value `-1` if this
is a compressed upload,
```

I think the condition should be `committedSize != expectedSize && (!chunker.isCompressed() || commitedSize != -1)`, as we are checking for write errors, right? (`!chunker.hasNext()` is redundant, so removed)

Reply:
Or we can split the condition if we are adding error message for compressed case.

Reply:
Refactored to add different error messages for compressed and uncompressed upload failures.